service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/AzureDatabaseConnectedTest.java

```diff
@@ -0,0 +1,202 @@
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.springframework.beans.factory.annotation.Autowired;
+
+// @Tag("azureConnected") - this test is tagged at the individual test level
+@TestInstance(Lifecycle.PER_CLASS)
+public class AzureDatabaseConnectedTest extends BaseAzureConnectedTest {
+
+  @Autowired private WorkspaceService workspaceService;
+  @Autowired private JobService jobService;
+  @Autowired private UserAccessUtils userAccessUtils;
+  @Autowired private ControlledResourceService controlledResourceService;
+  @Autowired private WsmResourceService wsmResourceService;
+  @Autowired private LandingZoneApiDispatch landingZoneApiDispatch;
+  @Autowired private SamService samService;
+
+  private Workspace sharedWorkspace;
+  private UUID workspaceUuid;
+
+  @BeforeAll
+  public void setup() throws InterruptedException {
+    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+    sharedWorkspace = createWorkspaceWithCloudContext(workspaceService, userRequest);
+    workspaceUuid = sharedWorkspace.getWorkspaceId();
+  }
+
+  @AfterAll
+  public void cleanup() {
+    // Deleting the workspace will also delete any resources contained in the workspace, including
+    // VMs and the resources created during setup.
+    workspaceService.deleteWorkspace(sharedWorkspace, userAccessUtils.defaultUserAuthRequest());
+  }
+
+  @Tag("azureConnected")
+  @Test
+  public void createAndDeleteAzureManagedIdAndDatabase() throws InterruptedException {
+    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+
+    var uamiResource = createManagedIdentity(userRequest);
+    var dbResource = createDatabase(userRequest, uamiResource);
+
+    // Verify that the resources we created
+    var resourceList = wsmResourceService.enumerateResources(workspaceUuid, null, null, 0, 100);
+    checkForResource(resourceList, uamiResource);
+    checkForResource(resourceList, dbResource);
+
+    // wait for azure to sync then make sure the resources actually exist
+    TimeUnit.SECONDS.sleep(5);
+    var actualUami =
+        getManagedIdentityFunction()
+            .apply(
+                azureTestUtils.getAzureCloudContext().getAzureResourceGroupId(),
+                uamiResource.getManagedIdentityName());
+    assertNotNull(actualUami);
+
+    var actualDatabase =
+        getDatabaseFunction()
+            .apply(
+                azureTestUtils.getAzureCloudContext().getAzureResourceGroupId(),
+                dbResource.getDatabaseName());
+    assertNotNull(actualDatabase);
+
+    deleteDatabase(userRequest, dbResource);
```

Comment:
It seems like the deletions should happen in a finally (or cleanup function). For instance, if only one of the 2 things fails to create, the other won't be deleted.

Reply:
the workspace is deleted in the `cleanup` function which should sweep everything up